src/aiida/transports/transport.py

```diff
@@ -39,15 +43,22 @@ def validate_positive_number(ctx, param, value):
     return value
 
 
-class Transport(abc.ABC):
-    """Abstract class for a generic transport (ssh, local, ...) contains the set of minimal methods."""
+def path_2_str(path: _TransportPath) -> str:
+    """Convert an instance of _TransportPath = Union[str, Path, PurePosixPath] instance to a string."""
+    # We could check if the path is a Path or PurePosixPath instance, but it's too much overhead.
+    return str(path)
+
+
+class _BaseTransport:
```

Comment:
no this is private. No one should inherent from this except 'AsyncTransport', 'BlockingTransport'.
Only 'AsyncTransport', 'BlockingTransport' are the public ones -- to be used to create a new plugin--

Reply:
I think it is problematic to have class like this, take the method `get_safe_open_interval` as example.

```python
    def get_safe_open_interval(self):
        """Get an interval (in seconds) that suggests how long the user should wait
        between consecutive calls to open the transport.  This can be used as
        a way to get the user to not swamp a limited number of connections, etc.
        However it is just advisory.
        If returns 0, it is taken that there are no reasons to limit the
        frequency of open calls.
        In the main class, it returns a default value (>0 for safety), set in
        the _DEFAULT_SAFE_OPEN_INTERVAL attribute of the class. Plugins should override it.
        :return: The safe interval between calling open, in seconds
        :rtype: float
        """
        return self._safe_open_interval
```

It says "Plugins should override it", then what is the point of define the method?